galaxy/management/commands/generate_galaxy_test_data.py

```diff
@@ -0,0 +1,288 @@
+                if (i / 10) % 10 == (i + j) % 10:
+                    u1 = self.users[i]
+                    u2 = self.users[j]
+                    self.logger.info(f"Making {u2} the godfather of {u1}")
+                    u1.godfathers.add(u2)
+
+    def make_club_memberships(self):
+        # This function makes multiple passes on all users to affect them some pseudo-random roles in some clubs.
+        # The multiple passes are useful to get some variations over who goes where.
+        # Each pass for each user has a chance to affect her to two different clubs, increasing a bit more the created chaos, while remaining purely deterministic.
+        for i in range(1, 11):  # users can be in up to 20 clubs
+            self.logger.info(f"Club membership, pass {i}")
+            for uid in range(
+                i, self.NB_USERS, i
+            ):  # Pass #1 will make sure every user is at least in one club
+                user = self.users[uid]
+                club = self.clubs[(uid + i**2) % self.NB_CLUBS]
+
+                start = self.now - timedelta(
+                    days=(((self.NB_USERS - uid) * i) // 110)
+                )  # older users were in clubs before newer users
+                end = start + timedelta(days=180)  # about one semester
+                self.logger.info(
+                    f"Making {user} a member of club {club} from {start} to {end}"
+                )
+                Membership(
+                    user=user,
+                    club=club,
+                    role=(uid + i) % 10 + 1,  # spread the different roles
+                    start_date=start,
+                    end_date=end,
+                ).save()
+
+            for uid in range(
+                10 + i * 2, self.NB_USERS, 10 + i * 2
+            ):  # Make a second affectation that will skip most users, to make a few citizen more important
+                user = self.users[uid]
+                club = self.clubs[(uid + i**3) % self.NB_CLUBS]
+
+                start = self.now - timedelta(
+                    days=(((self.NB_USERS - uid) * i) // 100)
+                )  # older users were in clubs before newer users
+                end = start + timedelta(days=180)  # about one semester
+                self.logger.info(
+                    f"Making {user} a member of club {club} from {start} to {end}"
+                )
+                Membership(
+                    user=user,
+                    club=club,
+                    role=((uid // 10) + i) % 10 + 1,  # spread the different roles
+                    start_date=start,
+                    end_date=end,
+                ).save()
+
+    def make_pictures(self):
+        # This function creates pictures for users to be tagged on later
+        self.picts = {}
+        for i in range(self.NB_USERS):
+            u = self.users[i]
+            # Create twice as many pictures as users
+            for j in [i, i**2]:
```

Comment:
There won't be users * 2 pictures, because the first sqrt(NB_USERS) ones will be overriden.
For example, the second user should have two pictures : 2 and 4. But when the images of the fourth user will be generated, picture 4 will then belong to the latter.

Reply:
Changed to use `i+NB_USERS`